app/controllers/users_controller.rb

```diff
@@ -0,0 +1,43 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+
+    @user = User.new(name: params[:user][:name], password: params[:user][:password])
+    if @user.save
+      login @user  
+      redirect_to @user
+
+    else 
+
+      redirect_to signup_path
+
+    end 
+
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+
+    events_separated =User.events_dates(@user.events_attended)
```

Comment:
It is better is you relly in the associations you already built instead of using a helper to get the past and upcoming events. For example, if you want the upcoming events attended by the current user you can use something like: `current_user.events_attended.upcoming_events`.